src/commands/generate/errorGenerator.ts

```diff
@@ -20,18 +20,20 @@ import { ImportDeclarationStructure } from "ts-simple-ast";
 import { ImportsVisitor, sortImports } from "./imports";
 import { SimpleAst } from "./simpleAst";
 import { TsReturnTypeVisitor } from "./tsReturnTypeVisitor";
+import { ITypeGenerationFlags } from "./typeGenerationFlags";
 import { doubleQuote, singleQuote } from "./utils";
 
 export function generateError(
     definition: IErrorDefinition,
     knownTypes: Map<string, ITypeDefinition>,
     simpleAst: SimpleAst,
+    typeGenerationFlags: ITypeGenerationFlags,
```

Comment:
its a shame that we have to plumb this down all over the place. Wish we constructed the various visitors in one place and then passed those around

Reply:
hmm yea it is not so bad but I can give it a go and create the visitors in the caller, and pass them down instead of the flags (in the top level `generate` function)